services/intersection-api/api/src/main/java/us/dot/its/jpo/ode/api/controllers/EventController.java

```diff
@@ -0,0 +1,1023 @@
+            list.add(MockEventGenerator.getSpatMessageCountProgressionEvent());
+            return ResponseEntity.ok(list);
+        } else {
+            Query query = spatMessageCountProgressionEventRepo.getQuery(intersectionID, startTime, endTime, latest);
+            return ResponseEntity.ok(spatMessageCountProgressionEventRepo.find(query));
+        }
+    }
+
+    @RequestMapping(value = "/events/spat_message_count_progression/count", method = RequestMethod.GET, produces = "application/json")
+    @PreAuthorize("hasRole('USER') || hasRole('ADMIN')")
+    public ResponseEntity<Long> countSpatMessageCountProgressionEvents(
+            @RequestParam(name = "intersection_id", required = false) Integer intersectionID,
+            @RequestParam(name = "start_time_utc_millis", required = false) Long startTime,
+            @RequestParam(name = "end_time_utc_millis", required = false) Long endTime,
+            @RequestParam(name = "full_count", required = false, defaultValue = "true") boolean fullCount,
+            @RequestParam(name = "test", required = false, defaultValue = "false") boolean testData) {
+
+        if (testData) {
+            return ResponseEntity.ok(1L);
+        } else {
+            Query query = spatMessageCountProgressionEventRepo.getQuery(intersectionID, startTime, endTime, false);
+
+            long count;
+
+            if (fullCount) {
+                count = spatMessageCountProgressionEventRepo.getQueryFullCount(query);
+            } else {
+                count = spatMessageCountProgressionEventRepo.getQueryResultCount(query);
+            }
+
+            log.debug("Found: {} SpatMessageCountProgressionEvents", count);
+            return ResponseEntity.ok(count);
+        }
+    }
+
+    @RequestMapping(value = "/events/map_message_count_progression", method = RequestMethod.GET, produces = "application/json")
+    @PreAuthorize("hasRole('USER') || hasRole('ADMIN')")
+    public ResponseEntity<List<MapMessageCountProgressionEvent>> findMapMessageCountProgressionEvents(
+            @RequestParam(name = "intersection_id", required = false) Integer intersectionID,
+            @RequestParam(name = "start_time_utc_millis", required = false) Long startTime,
+            @RequestParam(name = "end_time_utc_millis", required = false) Long endTime,
+            @RequestParam(name = "latest", required = false, defaultValue = "false") boolean latest,
+            @RequestParam(name = "test", required = false, defaultValue = "false") boolean testData) {
+
+        if (testData) {
+            List<MapMessageCountProgressionEvent> list = new ArrayList<>();
+            list.add(MockEventGenerator.getMapMessageCountProgressionEvent());
+            return ResponseEntity.ok(list);
+        } else {
+            Query query = mapMessageCountProgressionEventRepo.getQuery(intersectionID, startTime, endTime, latest);
+            return ResponseEntity.ok(mapMessageCountProgressionEventRepo.find(query));
```

Comment:
issue(blocking): executing query second time for logging purposes

Reply:
Extra query has been removed